### PR TITLE
GraphQL Subscriptions: Part 5

### PR DESCRIPTION
**Pull request by aklish**
Adds web socket support (JSR 356) and implementation of the [graphql-ws protocol](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md) (Apollo web socket protocol for consuming graphql subscriptions).

IT tests will come in another PR.

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: elide-graphql/src/main/java/com/yahoo/elide/graphql/subscriptions/websocket/SessionHandler.java [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+    }
+
+    protected void safeSendConnectionAck() {
+        ObjectMapper mapper = elide.getElideSettings().getMapper().getObjectMapper();
+        ConnectionAck ack = new ConnectionAck();
+
+        try {
+            sendMessage(mapper.writeValueAsString(ack));
+        } catch (JsonProcessingException e) {
+            safeClose(INTERNAL_ERROR);
```
Comment (moizarafat) [inline elide-graphql/src/main/java/com/yahoo/elide/graphql/subscriptions/websocket/SessionHandler.java:258]: should we log the error message for debugging?

File: elide-graphql/src/main/java/com/yahoo/elide/graphql/subscriptions/websocket/SessionHandler.java [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+    }
+
+    protected void safeSendPong() {
+        ObjectMapper mapper = elide.getElideSettings().getMapper().getObjectMapper();
+        Pong pong = new Pong();
+
+        try {
+            sendMessage(mapper.writeValueAsString(pong));
+        } catch (JsonProcessingException e) {
+            safeClose(INTERNAL_ERROR);
```
Comment (moizarafat) [inline elide-graphql/src/main/java/com/yahoo/elide/graphql/subscriptions/websocket/SessionHandler.java:269]: same comment as above.

File: elide-graphql/src/test/java/com/yahoo/elide/graphql/subscriptions/SubscriptionWebSocketTest.java [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+                .id("1")
+                .payload(Subscribe.Payload.builder()
+                        .query("subscription {book(topic: ADDED) {id title}}")
+                        .build())
+                .build();
+
+        endpoint.onMessage(session, mapper.writeValueAsString(subscribe));
+
+        String complete = "{\"id\":\"5d585eff-ed05-48c2-8af7-ad662930ba74\",\"type\":\"complete\"}";
+        //Complete complete = Complete.builder().id("1").build();
```
Comment (moizarafat) [inline elide-graphql/src/test/java/com/yahoo/elide/graphql/subscriptions/SubscriptionWebSocketTest.java:485]: Please remove commented code.

File: elide-standalone/src/main/java/com/yahoo/elide/standalone/config/ElideResourceConfig.java [hunk 4]
```diff
@@ -77,6 +78,134 @@
...
+                    ? new ElideStandaloneAsyncSettings() { } : settings.getAsyncProperties();
+            EntityManagerFactory entityManagerFactory = Util.getEntityManagerFactory(classScanner,
+                    settings.getModelPackageName(), asyncProperties.enabled(), settings.getDatabaseProperties());
+
+            EntityDictionary dictionary = settings.getEntityDictionary(injector, classScanner, dynamicConfiguration,
+                    settings.getEntitiesToExclude());
+
+            DataStore dataStore;
+
+
```
Comment (moizarafat) [inline elide-standalone/src/main/java/com/yahoo/elide/standalone/config/ElideResourceConfig.java:100]: Please remove extra new line.

**Review by rishi-aga — APPROVED** [review]

**Pull request merged**